it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/storage/GcsResourceManager.java

```diff
@@ -247,6 +295,20 @@ public void registerTempDir(String dirName) {
 
   @Override
   public synchronized void cleanupAll() {
+    // If bucket was created by manager, simply delete the bucket without worrying about other
+    // cleanups.
+    if (hasNonStaticBucket) {
+      boolean deleted = client.delete(bucket);
+      if (deleted) {
+        LOG.debug("Bucket '{}' was deleted", bucket);
+      } else {
+        LOG.warn("Bucket '{}' not deleted", bucket);
```

Comment:
Is it possible to cover this with a test?

Reply:
Done